tests/test_core.py

```diff
@@ -7961,20 +7965,18 @@ def setUp(self):
     os.chdir(self.get_dir()) # Ensure the directory exists and go there
 
     self.emcc_args = emcc_args[:]
-    Settings.load(self.emcc_args)
     Building.LLVM_OPTS = 0
-
-    Building.COMPILER_TEST_OPTS += [
+    Building.COMPILER_TEST_OPTS = [
         '-Werror', '-Wno-dynamic-class-memaccess', '-Wno-format',
         '-Wno-format-extra-args', '-Wno-format-security',
         '-Wno-pointer-bool-conversion', '-Wno-unused-volatile-lvalue',
         '-Wno-c++11-compat-deprecated-writable-strings',
         '-Wno-invalid-pp-token', '-Wno-shift-negative-value'
     ]
 
-    for arg in self.emcc_args:
-      if arg.startswith('-O'):
-        Building.COMPILER_TEST_OPTS.append(arg) # so bitcode is optimized too, this is for cpp to ll
+    # so bitcode is optimized too, this is for cpp to ll
+    if opt_level:
+      Building.COMPILER_TEST_OPTS.append(opt_level[-1])
```

Comment:
I read this wrong the first time, thinking `opt_level` would have to be `'-O2'` say, and `opt_level[-1]` would return `'2'`. Can we canonicalize `opt_level = opt_level[-1] if opt_level else None` earlier on?